### PR TITLE
Rov detection #83 issue

### PR DESCRIPTION
**Pull request by MAVRICK-1**
## Description
Solved the #83 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

@m-appel I had a pre-commit issue with the previous branch
so I closed that Pull request

### CODE REVIEW

File: iyp/crawlers/rovista/validating_rov.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    entries.append({'asn': entry['asn'], 'ratio': entry['ratio'], 'label': 'Validating RPKI ROV'})
+                else:
+                    entries.append({'asn': entry['asn'], 'ratio': entry['ratio'], 'label': 'Not Validating RPKI ROV'})
+
+            # Move to the next page
+            offset += 1
+            # Break the loop if there's no more data
+            if len(data) < batch_size:
+                break
+        logging.info('Pushing nodes to neo4j...\n')
```
Comment (m-appel) [inline iyp/crawlers/rovista/validating_rov.py:43]: Remove the `\n`

File: iyp/crawlers/rovista/validating_rov.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        # get ASNs and prefixes IDs
+        self.asn_id = self.iyp.batch_get_nodes_by_single_prop('AS', 'asn', asns)
+        tag_id_not_vali = self.iyp.get_node('Tag', {'label': 'Not Validating RPKI ROV'}, create=True)
```
Comment (m-appel) [inline iyp/crawlers/rovista/validating_rov.py:46]: `tag_id_not_valid` (`d` is missing)

File: iyp/crawlers/rovista/validating_rov.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        tag_id_vali = self.iyp.get_node('Tag', {'label': 'Validating RPKI ROV'}, create=True)
```
Comment (m-appel) [inline iyp/crawlers/rovista/validating_rov.py:47]: `tag_id_valid` (`d` is missing)

File: iyp/crawlers/rovista/validating_rov.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        # Compute links
+        links = []
+        for entry in entries:
+            asn_qid = self.asn_id[entry['asn']]
+            if entry['ratio'] > 0.5:
+                links.append({'src_id': asn_qid, 'dst_id': tag_id_vali, 'props': [self.reference, entry]})
```
Comment (m-appel) [inline iyp/crawlers/rovista/validating_rov.py:53]: We do not need the ASN and label in the relationship properties, so you can just do
```python
'props': [self.reference, {'ratio': entry['ratio']}]
```

File: iyp/crawlers/rovista/validating_rov.py [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+            else:
+                links.append({'src_id': asn_qid, 'dst_id': tag_id_not_vali, 'props': [self.reference, entry]})
```
Comment (m-appel) [inline iyp/crawlers/rovista/validating_rov.py:55]: Same as above

File: iyp/crawlers/rovista/validating_rov.py [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+
+        logging.info('Pushing links to neo4j...\n')
```
Comment (m-appel) [inline iyp/crawlers/rovista/validating_rov.py:57]: Remove the `\n`

**Review by m-appel — CHANGES_REQUESTED** [review]
Congratulations, you have a working crawler! :)

I've commented on a few nitpicks, once these are done this PR is good to be merged.

Btw. have you installed pre-commit locally? It should report the same problems as the GitHub runner when you commit, i.e., you do not have to wait for the output of the runner.

**Comment by MAVRICK-1** [comment]
@m-appel Thanks for Your Guidance , I made the requested changes , and Sry for the numerous commits , I am new to using pre-commit . In the Future , I will strive to minimize the frequency of commits

**Comment by MAVRICK-1** [comment]
> Congratulations, you have a working crawler! :)
> 
> I've commented on a few nitpicks, once these are done this PR is good to be merged.
> 
> Btw. have you installed pre-commit locally? It should report the same problems as the GitHub runner when you commit, i.e., you do not have to wait for the output of the runner.

I was using codespace for that pre-commit was not working ;-)

**Comment by MAVRICK-1** [comment]
> Congratulations, you have a working crawler! :)
> 
> I've commented on a few nitpicks, once these are done this PR is good to be merged.
> 
> Btw. have you installed pre-commit locally? It should report the same problems as the GitHub runner when you commit, i.e., you do not have to wait for the output of the runner.

Have installed it but it's showing 

```
pre-commit : The term 'pre-commit' is not recognized as the name of a cmdlet, function, 
script file, or operable program. Check the spelling of the name, or if a path was included, 
verify that the path is correct and try again.
```

**Review by m-appel — APPROVED** [review]
Thanks, looks good to me now!

**Pull request merged**